apollo-router/src/plugins/coprocessor/supergraph.rs

```diff
@@ -0,0 +1,881 @@
+    // Encode headers, body, status, context, sdl to create a payload
+    let headers_to_send = response_config
+        .headers
+        .then(|| externalize_header_map(&parts.headers))
+        .transpose()?;
+    let body_to_send = response_config
+        .body
+        .then(|| serde_json::to_value(&first).expect("serialization will not fail"));
+    let status_to_send = response_config.status_code.then(|| parts.status.as_u16());
+    let context_to_send = response_config.context.then(|| response.context.clone());
+    let sdl_to_send = response_config.sdl.then(|| sdl.clone().to_string());
+
+    let payload = Externalizable::supergraph_builder()
+        .stage(PipelineStep::SupergraphResponse)
+        .and_id(TraceId::maybe_new().map(|id| id.to_string()))
+        .and_headers(headers_to_send)
+        .and_body(body_to_send)
+        .and_context(context_to_send)
+        .and_status_code(status_to_send)
+        .and_sdl(sdl_to_send.clone())
+        .build();
+
+    // Second, call our co-processor and get a reply.
+    tracing::debug!(?payload, "externalized output");
+    let guard = response.context.enter_active_request();
+    let co_processor_result = payload.call(http_client.clone(), &coprocessor_url).await;
+    drop(guard);
+    tracing::debug!(?co_processor_result, "co-processor returned");
+    let co_processor_output = co_processor_result?;
+
+    validate_coprocessor_output(&co_processor_output, PipelineStep::SupergraphResponse)?;
+
+    // Third, process our reply and act on the contents. Our processing logic is
+    // that we replace "bits" of our incoming response with the updated bits if they
+    // are present in our co_processor_output. If they aren't present, just use the
+    // bits that we sent to the co_processor.
+    let new_body: crate::response::Response = match co_processor_output.body {
+        Some(value) => serde_json::from_value(value)?,
+        None => first,
+    };
+
+    //response.response = http::Response::from_parts(parts, stream::once(future::ready(new_body)));
+
+    if let Some(control) = co_processor_output.control {
+        parts.status = control.get_http_status()?
+    }
+
+    if let Some(context) = co_processor_output.context {
+        for (key, value) in context.try_into_iter()? {
+            response
+                .context
+                .upsert_json_value(key, move |_current| value);
+        }
+    }
+
+    if let Some(headers) = co_processor_output.headers {
+        parts.headers = internalize_header_map(headers)?;
+    }
+
+    // Now break our co-processor modified response back into parts
+    //let (parts, body) = response.response.into_parts();
+
+    // Clone all the bits we need
+    let context = response.context.clone();
+    let map_context = response.context.clone();
+
+    // Map the rest of our body to process subsequent chunks of response
+    let mapped_stream = rest
+        .then(move |deferred_response| {
```

Comment:
```suggestion
        .then(move |response_chunk| {
```
deferred is a bit ambiguous

Reply:
"response chunk" feels more like a router service concern. At the supergraph level, a deferred response is the correct name. We can have a primary response then a stream of deferred responses (each of them is in the shape of a GraphQL response)